src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/reports/EnvelopeCountSummaryRepository.java

```diff
@@ -15,18 +15,26 @@
     @Query(
         nativeQuery = true,
         value = "SELECT\n"
-            + "  container,\n"
-            + "  date,\n"
+            + "  first_events.container AS container,\n"
+            + "  date(first_events.createdat) AS date,\n"
             + "  count(*) AS received,\n"
-            + "  SUM(CASE WHEN event IN ('DOC_FAILURE', 'FILE_VALIDATION_FAILURE', 'DOC_SIGNATURE_FAILURE') THEN 1 ELSE 0 END) AS rejected\n"
+            + "  SUM(CASE WHEN rejection_events.id IS NOT NULL THEN 1 ELSE 0 END) AS rejected\n"
             + "FROM (\n"
             + "  SELECT DISTINCT on (container, zipfilename)\n"
-            + "  container, zipfilename, date(createdat), event\n"
+            + "    container, zipfilename, createdat\n"
             + "  FROM process_events\n"
-            + "  ORDER BY container, zipfilename, date(createdat) ASC\n"
+            + "  ORDER BY container, zipfilename, createdat ASC\n"
             + ") AS first_events\n"
-            + "GROUP BY container, date\n"
-            + "HAVING date = :date"
+            + "LEFT JOIN (\n"
+            + "  SELECT DISTINCT on (container, zipfilename)\n"
+            + "    id, container, zipfilename\n"
+            + "  FROM process_events\n"
+            + "  WHERE event IN ('DOC_FAILURE', 'FILE_VALIDATION_FAILURE', 'DOC_SIGNATURE_FAILURE')\n"
+            + ") AS rejection_events\n"
+            + "ON rejection_events.container = first_events.container\n"
+            + "AND rejection_events.zipfilename = first_events.zipfilename\n"
+            + "GROUP BY first_events.container, date(first_events.createdat)\n"
+            + "HAVING date(first_events.createdat) = :date\n"
```

Comment:
There are two things not optimal in this bit. Partly it was like that in the previous version, but worth improving, as we're doing a full database scan in this query.
* We don't need to calculate the date in this query, at all, because we receive it in the parameter
* Instead of HAVING we should have a condition in WHERE in first_events subquery - this way the join will be far less costly. Currently, the whole table data is involved in the query and the more records we add the more of a problem this will become.

Reply:
I also realised we have a missing index - for createdat.